src/js/controllers/targets.js

```diff
@@ -18,9 +18,20 @@ angular.module('app').controller("TargetsController", ["$scope", "Kong", "$route
 
   $scope.loadMore = function() {
     var page = '/upstreams/' + upstream.id + '/targets';
-    if ($scope.active) {
-      page += "/active";
+
+    // For Kong 0.12, the APIs for accessing active and all targets have changed
+    // https://getkong.org/docs/0.12.x/admin-api/#list-target
+    if (env.kong_version.startsWith("0.12")) {
+      if (!$scope.active) {
+        page += "/all";
+      }
+    } else {
+      // For Kong versions other than 0.12
+      if ($scope.active) {
+        page += "/active";
+      }
```

Comment:
Problem with this logic is that when a new minor version of Kong will be available, kong dashboard will use the `/active` endpoint back again. I doubt that Kong 0.13 will go back to Kong 0.11 behavior for this endpoint.

How about this kind of logic?
```
if (["0.9.", "0.10", "0.11"].includes(env.kong_version.substring(0,4)) {
    page += $scope.active ? "/active" : "";
} else {
    page += !$scope.active ? "/all" : "";
}
```

Reply:
Okay. Make sense.